versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/RocksFragment.java

```diff
@@ -127,4 +139,77 @@ public void evaluate(Function function) throws ConditionFailedException {
   private Entity keysAsEntityList(List<Key> keys) {
     return Entity.ofList(keys.stream().map(k -> Entity.ofList(k.getElements().stream().map(Entity::ofString))));
   }
+
+  @Override
+  protected void remove(String fieldName, ExpressionPath.PathSegment path) {
+    if (new PathPattern().anyPosition().matches(path)) {
+      final int i = path.asPosition().getPosition();
+      fragmentBuilder.removeKeys(i);
+    } else if (new PathPattern().anyPosition().anyPosition().matches(path)) {
+      final int outerIndex = path.asPosition().getPosition();
+      final int innerIndex = path.getChild().get().asPosition().getPosition();
+
+      final List<String> updatedKeys = new ArrayList<>(fragmentBuilder.getKeys(outerIndex).getElementsList());
+      updatedKeys.remove(innerIndex);
+
+      fragmentBuilder.setKeys(path.asPosition().getPosition(), ValueProtos.Key
+          .newBuilder()
+          .addAllElements(updatedKeys)
+      );
+    } else {
+      throw new UnsupportedOperationException("Invalid path for remove");
+    }
+  }
+
+  @Override
+  protected boolean fieldIsList(String fieldName, ExpressionPath.PathSegment childPath) {
+    if (!KEY_LIST.equals(fieldName)) {
+      return false;
+    }
+
+    return new PathPattern().matches(childPath) || new PathPattern().anyPosition().matches(childPath);
+  }
+
+  @Override
+  protected void appendToList(String fieldName, ExpressionPath.PathSegment childPath, List<Entity> valuesToAdd) {
+    if (new PathPattern().matches(childPath)) {
+      fragmentBuilder.addAllKeys(valuesToAdd.stream().map(v -> ValueProtos.Key
+          .newBuilder()
+          .addAllElements(v.getList().stream().map(Entity::getString).collect(Collectors.toList()))
+          .build()
+      ).collect(Collectors.toList()));
```

Comment:
You should be able to avoid intermediate list creation by adding incrementally and not using addAllKeys.

Reply:
Please apply this to all applicable locations.

Reply:
Updated.